### DISCUSSION TITLE
[QnA] Need help with window rules.

### DISCUSSION THREAD

**Discussion by H1manshu-Soni** [prasanthrangan/hyprdots#920] [Q&A] [answered]
I want certain apps to always float whenever opened. Like for example clock app which has no need to be tiled or calculator which just have a quick use and then close.

I tried to find solutions online but couldnt get it working if anyone can help me with this let me know.

**Comment by kRHYME7** [comment]
This? 
https://wiki.hyprland.org/Configuring/Window-Rules/


Can you share your attempts in configuring the windows rules?

**Reply by H1manshu-Soni** [reply]
first place I went to but couldnt get it working :-(

**Reply by kRHYME7** [reply]
Have you tried this tip?

![Screenshot_2024-02-21-00-43-59-185_com.brave.browser-edit.jpg](https://github.com/prasanthrangan/hyprdots/assets/53417443/f11fa1f6-7b68-45f5-8987-9ecfa579b93c)

**Comment by H1manshu-Soni** [comment]
Sure, I added two lines in ~/.config/hypr/windowrules.conf 

windowrulev2 = opacity 0.80 0.80,class:^(gnome-clocks)$
windowrulev2 = float,class:^(gnome-clocks)$

**Reply by kRHYME7** [reply]
how did you find out that the class is ``` gnome-clocks ``` ?


Try to change the class with this
 ```  org.gnome.clocks ```

**Reply by H1manshu-Soni** [reply]
well i didnt i just tried with that,
the org.gnome.clocks doesnt seem to work

**Reply by kRHYME7** [reply, marked as answer]
While the gnome-clock is running 
try to run ``` hyprctl clients | grep class ```
what is the output?

**Reply by H1manshu-Soni** [reply]
class : org.gnome.clocks

**Reply by H1manshu-Soni** [reply]
i was using ' - ' instead of ' . '  it works now thanks.
